Allow `Person` to have multiple `Survey` objects

ryanlml opened the pull request:
Resolves #99. There were a lot more changes than I foresaw. I have changed some of the behaviour for the following commands:

- `add` now parses all `s/` prefixes.
```
add n/John Doe ... s/Survey A s/Survey B
> New person added: John Doe ... Surveys: [Survey A][Survey B]
 ```
- `view s/<KEYWORD>` no longer lists persons whose surveys contain the word `<KEYWORD>`. Instead, it lists person who has `<KEYWORD>` as a survey name. This may be improved in #107.
```
view s/Academic
> 0 persons listed!

view s/Academic Survey
> 1 persons listed!
> John Doe ... Surveys: [Academic Survey][Environment Survey]
```
- `delete s/<survey name>` now deletes persons whose surveys contain `<survey name>`, even if they contain other surveys. This is perhaps undesirable; we may want to remove the specified survey from the person rather than deleting the person altogether. Thoughts @deepimpactmir?
```
view s/Academic Survey
> 1 persons listed!
> John Doe ... Surveys: [Academic Survey][Environment Survey]

delete s/Academic Survey
> Deleted person [John Doe ... Surveys: [Academic Survey][Environment Survey]]
```
- `clone` should work as per usual

Please help test your commands! There may be other cases that messes up when multiple surveys are involved, but these are the ones I found.

Also changed the Ui to show multiple surveys, similar to `tags`.

ryanlml committed:
Add functionality for `Person` class to keep track of multiple `Survey` objects
Update user interface so that the survey names look better

ryanlml committed:
Fix coding style violations

ryanlml committed:
Change tests to fix new functionalities

ryanlml committed:
Add missing Javadoc

ryanlml committed:
Change tests to fit new functionalities

KeithPJX commented on src/main/java/seedu/address/storage/JsonAdaptedPerson.java:
```diff
@@ -50,7 +50,8 @@ public JsonAdaptedPerson(@JsonProperty("name") String name, @JsonProperty("phone
             @JsonProperty("gender") String gender, @JsonProperty("birthdate") String birthdate,
             @JsonProperty("race") String race, @JsonProperty("religion") String religion,
-            @JsonProperty("survey") String survey, @JsonProperty("tagged") List<JsonAdaptedTag> tagged) {
+            @JsonProperty("survey") List<JsonAdaptedSurvey> survey,
```
Wondering if this should be changed to surveys for both the name of the List and the JsonProperty. Will have to refactor all the JSON files though.

ryanlml replied:
You're right, `surveys` is better. Shouldn't be too much of a problem factoring JSON files.

ryanlml replied:
Renamed `survey` to `surveys` as suggested.

KeithPJX reviewed (approved):
LGTM. Pointed out a coding standard violation. Also, I agree that delete should just delete the survey mentioned if person has multiple surveys, if the person only has 1 survey then delete the whole person.

deepimpactmir commented on src/main/java/seedu/address/storage/JsonAdaptedPerson.java:
```diff
@@ -59,14 +60,16 @@ public JsonAdaptedPerson(@JsonProperty("name") String name, @JsonProperty("phone
 
     /**
-     * Converts a given {@code Person} into this class for Jackson use.
+     * Converts a given {@code PJsonAdaptedTagerson} into this class for Jackson use.
```
Not sure what this is supposed to mean. Probably a typo?

ryanlml replied:
Thanks for pointing it out. Will fix them now.

ryanlml replied:
Fixed typo.

deepimpactmir reviewed (approved):
Thanks for all the changes! :). I agree with your sentiments on removing just the survey on persons with more than one survey. I should be able to open a PR on that soon.

ryanlml committed:
Implement suggestions and fixes by @KeithPJX and @deepimpactmir
Rename `survey` to `surveys` in `JsonAdaptedPerson` class
Rename all `survey` to `surveys` in .json files
Fix typo in `JsonAdaptedPerson`
